### PR TITLE
[dev] Have CLI and GUI builds share artifacts

### PR DESCRIPTION
**Pull request by oh-yes-0-fps**

### CODE REVIEW

**Review by calcmogul — CHANGES_REQUESTED** [review]
`pnpm dev --release` breaks locally with this:
```
thread '<unnamed>' panicked at src/interface/rust/desktop.rs:52:69:
failed to rename app: failed to rename `/home/tav/git/Choreo/target/debug/choreo` to `/home/tav/git/Choreo/target/debug/choreo`

Caused by:
    No such file or directory (os error 2)
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
It's trying to copy a file from the debug folder when the --release flag was passed. That used to work.

**Comment by oh-yes-0-fps** [comment]
Ill look into it but why are you running release on dev?

**Comment by calcmogul** [comment]
`pnpm dev` fails with the same error message.

**Comment by oh-yes-0-fps** [comment]
this works on windows again, ill see what I can do

**Comment by oh-yes-0-fps** [comment]
hmm, i think it might be easier to override rust analyzer output location than it is to override tauri, seems there is some behavior that's not the same on linux that causes using a different output dir to break the loading

File: .github/workflows/choreo.yml [hunk 1]
```diff
@@ -16,23 +16,23 @@ jobs:
         include:
           - artifact-name: Windows-x86_64
             os: windows-2022
-            tauri-build-flags: --target x86_64-pc-windows-msvc
+            tauri-build-flags: --target x86_64-pc-windows-msvc -v -- --workspace
```
Comment (calcmogul) [inline .github/workflows/choreo.yml:19]: Are the -v flags intentional?

**Review by calcmogul — CHANGES_REQUESTED** [review]
I ran `pnpm install`, then `pnpm dev --release`:
```
> choreo@2025.0.0-beta dev /home/tav/git/Choreo
> vite "--release"

file:///home/tav/git/Choreo/node_modules/.pnpm/vite@5.4.3_@types+node@22.5.4/node_modules/vite/dist/node/cli.js:444
          throw new CACError(`Unknown option \`${name.length > 1 ? `--${name}` : `-${name}`}\``);
                ^

CACError: Unknown option `--release`
    at Command.checkUnknownOptions (file:///home/tav/git/Choreo/node_modules/.pnpm/vite@5.4.3_@types+node@22.5.4/node_modules/vite/dist/node/cli.js:444:17)
    at CAC.runMatchedCommand (file:///home/tav/git/Choreo/node_modules/.pnpm/vite@5.4.3_@types+node@22.5.4/node_modules/vite/dist/node/cli.js:642:13)
    at CAC.parse (file:///home/tav/git/Choreo/node_modules/.pnpm/vite@5.4.3_@types+node@22.5.4/node_modules/vite/dist/node/cli.js:581:12)
    at file:///home/tav/git/Choreo/node_modules/.pnpm/vite@5.4.3_@types+node@22.5.4/node_modules/vite/dist/node/cli.js:914:5
    at ModuleJob.run (node:internal/modules/esm/module_job:234:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:473:24)

Node.js v20.17.0
 ELIFECYCLE  Command failed with exit code 1.
```
`pnpm dev` hangs:
```
[tav@myriad Choreo]$ pnpm dev

> choreo@2025.0.0-beta dev /home/tav/git/Choreo
> vite


  VITE v5.4.3  ready in 139 ms

  ➜  Local:   http://localhost:1420/
  ➜  Network: use --host to expose
  ➜  press h + enter to show help
```

File: src-tauri/tauri.conf.json [hunk 2]
```diff
@@ -53,7 +54,7 @@
         "minimumSystemVersion": "13.3",
         "signingIdentity": "-"
       },
-      "externalBin": ["../cli/choreo-cli"]
```
Comment (calcmogul) [inline src-tauri/tauri.conf.json:56]: Does this mean the cli folder no longer needs to be in .gitignore?

File: .github/workflows/choreo.yml (hunk 1 above)
Comment (oh-yes-0-fps) [reply .github/workflows/choreo.yml:19]: -v flags add extra info theres not really a point not having

File: .vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+    "rust-analyzer.check.overrideCommand": [
+        "cargo", "clippy", "--target-dir", "./target-clippy"
+    ],
+}
```
Comment (calcmogul) [inline .vscode/settings.json:5]: Add newline to end-of-file.

**Review by calcmogul — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 4]
```diff
@@ -26,7 +27,7 @@ dist-ssr
 # Choreo
 src-tauri/target
 target
-cli
+target-clippy
```
Comment (calcmogul) [inline .gitignore:30]: Delete this.

**Review by calcmogul — CHANGES_REQUESTED** [review]

**Review by calcmogul — APPROVED** [review]

**Pull request merged**